pkg/settlement/swap/chequebook/init.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chequebook
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/storage"
+)
+
+const chequebookKey = "chequebook"
+
+// Init initialises the chequebook service
+func Init(
+	ctx context.Context,
+	chequebookFactory Factory,
+	stateStore storage.StateStorer,
+	logger logging.Logger,
+	swapInitialDeposit uint64,
+	transactionService TransactionService,
+	swapBackend Backend,
+	overlayEthAddress common.Address,
+	simpleSwapBindingFunc SimpleSwapBindingFunc,
+	erc20BindingFunc ERC20BindingFunc) (chequebookService Service, err error) {
+	// verify that the supplied factory is valid
+	err = chequebookFactory.VerifyBytecode(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	erc20Address, err := chequebookFactory.ERC20Address(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var chequebookAddress common.Address
+	err = stateStore.Get(chequebookKey, &chequebookAddress)
+	if err != nil {
+		if err != storage.ErrNotFound {
+			return nil, err
+		}
+		// if we don't yet have a chequebook, deploy a new one
+		logger.Info("deploying new chequebook")
+
+		chequebookAddress, err = chequebookFactory.Deploy(ctx, overlayEthAddress, big.NewInt(0))
+		if err != nil {
+			return nil, err
+		}
+
+		logger.Infof("deployed chequebook at address %x", chequebookAddress)
+
+		// save the address for later use
+		err = stateStore.Put(chequebookKey, chequebookAddress)
+		if err != nil {
+			return nil, err
+		}
+
+		chequebookService, err = New(swapBackend, transactionService, chequebookAddress, erc20Address, overlayEthAddress, simpleSwapBindingFunc, erc20BindingFunc)
+		if err != nil {
+			return nil, err
+		}
+
+		if swapInitialDeposit != 0 {
+			logger.Info("depositing into new chequebook")
```

Comment:
Depositing <amount> into new chequebook

Reply:
Aha. Github removed part of my comment. I wanted to write `depositing <amount> into new chequebook`. The capitalization was not the point (and should indeed not be capitalized).